### PR TITLE
[LAD] Updates for Linux diagnostic-related changes

### PR DESCRIPTION
**Pull request by hosungsmsft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-vm/tests/test_vm_commands.py [hunk 1]
```diff
@@ -858,12 +858,12 @@ def test_diagnostics_extension_install(self):
...
 
     def body(self):
-        storage_key = '123'  # use junk keys, do not retrieve real keys which will get into the recording
+        storage_sastoken = '123'  # use junk keys, do not retrieve real keys which will get into the recording
         _, protected_settings = tempfile.mkstemp()
         with open(protected_settings, 'w') as outfile:
             json.dump({
                 "storageAccountName": self.storage_account,
-                "storageAccountKey": storage_key,
+                "storageAccountSasToken": storage_sastoken,
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/tests/test_vm_commands.py]: to confirm, using `storageAccountKey` should still work right?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py [hunk 2]
```diff
@@ -278,248 +278,702 @@ def get_default_diag_config(is_windows):
     else:
         return {
             "ladCfg": {
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py]: This is a data file in nutshell, to get CI passing, it is fine to just disable the flake8 errors by putting `# flake8: noqa` at the top. This way we can just copy and paste new config moving forward

File: src/command_modules/azure-cli-vm/tests/test_vm_commands.py (hunk 1 above)
Comment (hosungsmsft) [inline src/command_modules/azure-cli-vm/tests/test_vm_commands.py]: Actually, no. We were pushed from the Azure Security team not to use a full storage shared access key, and storageAccountKey is now fully deprecated. That's one of the reasons why LAD 3.0 can't be backward-compatible with LAD 2.3.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py (hunk 2 above)
Comment (hosungsmsft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py]: Thanks. I suppose this is about the first two Travis CI failures? Added.

I'm looking at the third failure for VMSS, and include a fix for that in the next iteration.

File: src/command_modules/azure-cli-vm/tests/test_vm_commands.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/tests/test_vm_commands.py]: Good to know we might need to add a warning for this. Don't worry, i can do that and let you take a look

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py]: This is a recorded test. Once you consolidate the test scenario, just delete the [recording](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/tests/recordings/test_diagnostics_extension_install.yaml) and rerun the test to produce new recordings. Let me know if you need more information

Comment (hosungsmsft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_diagnostics_templates.py]: Thanks for your info. Followed your lead and regenerated the new recording and pushed the 3rd commit with that change.

**Comment by yugangw-msft** [comment]
@hosungsmsft, please submit a new PR with needed help change

**Pull request merged**